x-pack/plugins/observability/server/services/slo/tasks/orphan_summary_cleanup_task.ts

```diff
@@ -0,0 +1,246 @@
+import {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { AggregationsCompositeAggregateKey } from '@elastic/elasticsearch/lib/api/types';
+import { ALL_SPACES_ID } from '@kbn/spaces-plugin/common/constants';
+import { ObservabilityConfig } from '../../..';
+import { SLO_SUMMARY_DESTINATION_INDEX_PATTERN } from '../../../../common/slo/constants';
+import { StoredSLO } from '../../../domain/models';
+import { SO_SLO_TYPE } from '../../../saved_objects';
+
+export const TASK_TYPE = 'SLO:ORPHAN_SUMMARIES-CLEANUP-TASK';
+
+export const getDeleteQueryFilter = (
+  sloSummaryIdsToDelete: Array<{ id: string; revision?: number }>
+) => {
+  return sloSummaryIdsToDelete.map(({ id, revision }) => {
+    return {
+      bool: {
+        must: [
+          {
+            term: {
+              'slo.id': id,
+            },
+          },
+          {
+            term: {
+              'slo.revision': Number(revision),
+            },
+          },
+        ],
+      },
+    };
+  });
+};
+
+export class SloOrphanSummaryCleanupTask {
+  private abortController = new AbortController();
+  private logger: Logger;
+  private taskManager?: TaskManagerStartContract;
+  private soClient?: SavedObjectsClientContract;
+  private esClient?: ElasticsearchClient;
+  private config: ObservabilityConfig;
+
+  constructor(taskManager: TaskManagerSetupContract, logger: Logger, config: ObservabilityConfig) {
+    this.logger = logger;
+    this.config = config;
+
+    taskManager.registerTaskDefinitions({
+      [TASK_TYPE]: {
+        title: 'SLO Definitions Cleanup Task',
+        timeout: '3m',
+        maxAttempts: 1,
+        createTaskRunner: ({ taskInstance }: { taskInstance: ConcreteTaskInstance }) => {
+          return {
+            run: async () => {
+              return this.runTask();
+            },
+
+            cancel: async () => {
+              this.abortController.abort('[SLO] Definitions clean up Task timed out');
+            },
+          };
+        },
+      },
+    });
+  }
+
+  runTask = async () => {
+    const runAt = new Date().toISOString();
+
+    if (this.soClient && this.esClient) {
+      let searchAfterKey: AggregationsCompositeAggregateKey | undefined;
+
+      do {
+        const { sloSummaryIds, searchAfter } = await this.fetchSloSummariesIds(searchAfterKey);
+
+        if (sloSummaryIds.length === 0) {
+          return { state: { lastRunAt: runAt } };
+        }
+
+        searchAfterKey = searchAfter;
+        const sloDefinitions = (
+          await this.soClient.bulkGet<Pick<StoredSLO, 'id' | 'revision'>>(
+            sloSummaryIds.map(({ id }) => ({
+              id,
+              type: SO_SLO_TYPE,
+            })),
+            {
+              namespace: ALL_SPACES_ID,
+            }
+          )
+        ).saved_objects.filter((so) => !so.error);
```

Comment:
Unfortunately we cannot use bulkGet because `SLO ID !== SO ID`.
We have to use the find api, with a filter on `slo.attributes.id:(id1 or id2 or ... or id1000)`. We do something like that in the [slo_repository.findAllById()](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability/server/services/slo/slo_repository.ts#L88-L105)